detox/e2e/test/messaging/channel_link.e2e.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// *******************************************************************
+// - [#] indicates a test step (e.g. # Go to a screen)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element testID when selecting an element. Create one if none.
+// *******************************************************************
+
+import {ChannelScreen} from '@support/ui/screen';
+import {
+    Channel,
+    Post,
+    Setup,
+} from '@support/server_api';
+import {timeouts, wait} from '@support/utils';
+
+describe('Channel Link', () => {
+    const {
+        channelNavBarTitle,
+        goToChannel,
+        openReplyThreadFor,
+        postMessage,
+    } = ChannelScreen;
+    let testChannel;
+    let testTeam;
+    let townSquareChannel;
+
+    beforeAll(async () => {
+        const {user, team, channel} = await Setup.apiInit();
+        testChannel = channel;
+        testTeam = team;
+
+        ({channel: townSquareChannel} = await Channel.apiGetChannelByName(team.id, 'town-square'));
+
+        // # Open channel screen
+        await ChannelScreen.open(user);
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    it('MM-T178 should be able to open channel by tapping on channel link from main channel', async () => {
+        // # Post a channel link
+        await goToChannel(townSquareChannel.display_name);
+        const channelLink = `http://localhost:8065/${testTeam.name}/channels/${testChannel.name}`;
```

Comment:
I'd recommend to use `siteUrl`?

Reply:
I tried adding `import testConfig from '@support/test_config';` but TSC is complaining

```
Checking for TSC
ERROR: Check TSC hints.
app/store/mmkv_adapter.ts(94,13): error TS2322: Type 'boolean | undefined' is not assignable to type 'Promise<boolean>'.
  Type 'undefined' is not assignable to type 'Promise<boolean>'.
```

I think for the boolean `siteUrl: process.env.SITE_URL || 'http://localhost:8065',`

I'm keeping the original for now as I'm not sure how to resolve it.

Reply:
Done